jobbergate-api/jobbergate_api/apps/services.py

```diff
@@ -248,22 +281,26 @@ def build_list_query(
             query = query.where(getattr(self.model_type, key) == value)
         if user_email:
             query = query.where(self.model_type.owner_email == user_email)
+        if not include_archived:
+            query = query.where(not_(self.model_type.is_archived))
         if search:
             require_condition(
                 hasattr(self.model_type, "searchable_fields"),
-                f"{self.model_type.__tablename__} does not support search",
+                f"{self.name} does not support search",
                 raise_exc_class=ServiceError,
                 raise_kwargs=dict(status_code=status.HTTP_405_METHOD_NOT_ALLOWED),
             )
             query = query.where(search_clause(search, self.model_type.searchable_fields()))
         if sort_field:
             require_condition(
                 hasattr(self.model_type, "sortable_fields"),
-                f"{self.model_type.__tablename__} does not support sort",
+                f"{self.name} does not support sort",
                 raise_exc_class=ServiceError,
                 raise_kwargs=dict(status_code=status.HTTP_405_METHOD_NOT_ALLOWED),
             )
             query = query.order_by(sort_clause(sort_field, self.model_type.sortable_fields(), sort_ascending))
+        if eager_join and self.parent_model_link is not None:
+            query.options(joinedload(self.parent_model_link, innerjoin=innerjoin))
```

Comment:
I wonder if we could add a `parent_model_link` attribute on the model itself. That way, the model can specify if it has a parent link or not. It seems odd to pass this to the service in this way.

Reply:
I've refactored it to a class method on the models, so there is no need to set it at the service level.